src/modules/mentors/mentors.controller.ts

```diff
@@ -46,7 +46,8 @@ export class MentorsController {
   @Get()
   @UsePipes(new PaginationPipe())
   async index(@Req() request: Request, @Query() filters: MentorFiltersDto) {
-    const data = await this.mentorsService.findAll(filters, !!request.user);
+    const userId: string = request.user.auth0Id;
```

Comment:
`req.user` would be `undefined` if it's an unauthenticated request. We should do something like the following.

```
const userId: string = request.user?.auth0Id || '';
```

Do you mind writing an E2E test covering this case?

Reply:
Woof.  Great catch.  

We shouldn't need to coalesce the string to an empty string, but you're totally right that this would cause an error.  Thank you for catching that!

Reply:
Done.